### ISSUE TITLE
cursor invisible on empty lines in chrome 62

### ISSUE THREAD

**Issue by kay999** [codemirror/CodeMirror#5056]
Cursor is visible in lines with some text, but invisible for blank lines. This fixes it:

.CodeMirror-cursor {
	width: 1px !important;
}

**Comment by marijnh** [comment]
I don't see this happening on Chrome 62 Linux. Do you see it on the demos on the website?

**Comment by kay999** [comment]
Yes, also on demos on the website. But only seem to happen on OS X for certain page-zoom factors < 1 (for example page zoom 80%).

**Comment by davidroman0O** [comment]
I had the same problem with my cursor at the beginning of a line, the css code of @kay999 fixed it!
thx!

**Comment by adrianheine** [comment]
Is this issue still present in Chrome > 62?

**Comment by ferrao** [comment]
@adrianheine, I confirm this issue on `Version 67.0.3396.99 (Official Build) (64-bit)`

**Comment by marijnh** [comment]
At a small enough zoom level (50%) I can reproduce this. Our cursor is an element with a 1 pixel border, and it appears that at some point Chrome just rounds that down to zero (not just on empty lines, also in some -- but not all -- other positions).

So that's unfortunate, but I don't know how to work around it without making the cursor fatter in all circumstances (which isn't desirable).

**Comment by adrianheine** [comment]
This is one of many issues which are difficult to solve with the fundamental approach currently taken by CodeMirror.

We are working on a rewrite (CodeMirror 6) that will overhaul input handling and should address this issue, and we are currently raising money for this work: See [the announcement](https://codemirror.net/6/) for more information about the rewrite and a demo.

Note that CodeMirror 6 is by no means stable or usable in production, yet. It is highly unlikely that we pick up this issue for CodeMirror 5, though.

**Comment by Zach10za** [comment]
I know this is an older thread but a better solution (and one that worked for me) is just to change the styling of the cursor from:
```css
.CodeMirror-cursor {
  border-left: 1px solid black;
  border-right: none;
  width: 0;
}
```
to
```css
.CodeMirror-cursor {
  border-left: thin solid black;
  border-right: none;
  width: 0;
}
```

The real problem is the browser is trying to calculate a sub-pixel border-width for odd resolutions/zoom levels. The css `border-width: thin` property seems to ignore the calculation and always remain 1px.

I couldn't find much on browser compatibility but I'd be happy to look more into it and make a PR if that would be helpful for you guys.

Here is the relevant line.
https://github.com/codemirror/CodeMirror/blob/17457466015551c7bf9292896baaa69e738f0d49/lib/codemirror.css#L46

**Comment by marijnh** [comment]
That's a useful trick, but at this point, unless there is some evidence that `thin` will usually end up being around a pixel in width, I'd be worried about changing the default style and breaking someone's setup.

**Comment by aw33598** [comment]
@kay999 I am facing same problem but where to put this code ?

**Comment by ferrao** [comment]
@kay999 your workaround breaks fat cursor mode, use this instead:

```
:not(.cm-fat-cursor) .CodeMirror-cursor {
    width: 1px !important;
}
```

**Comment by yuyuyu8818** [comment]
In my case, I needed
```
:not(.cm-fat-cursor) .CodeMirror-cursor {
    width: 1px !important;
    height: 18px !important;
}
```
environment
- Electron v8.2.0
- codemirror 5.52.2
- Manjaro Linux
- react v16.13.1
- codemirror used in react-bootstrap Tab container

question
- How is cursor height calculated?

**Comment by EricRabil** [comment]
My solution was to give the cursor a `margin-left` to shove it into view, because it is only ever one pixel off (in my environment, at least).
```css
.CodeMirror-cursor {
  margin-left: 1px;
}
```
